pkg/test/framework/components/istio/operator.go

```diff
@@ -367,15 +367,29 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 		}
 	}
 
+	// Need to determine if there is a setting to watch cluster secret in config cluster
+	// or in external cluster. The flag is named LOCAL_CLUSTER_SECERT_WATCHER and set as
+	// an environment variable for istiod.
+	watchLocalNamespace := false
+	if istioctlConfigFiles.operatorSpec != nil && istioctlConfigFiles.operatorSpec.Values != nil {
+		localClusterSecretWatcher := GetConfigValue("pilot.env.LOCAL_CLUSTER_SECERT_WATCHER",
+			istioctlConfigFiles.operatorSpec.Values.Fields)
+		if localClusterSecretWatcher.GetStringValue() == "true" && i.isExternalControlPlane() {
+			watchLocalNamespace = true
+		}
+	}
+
+	if ctx.Clusters().IsMulticluster() {
+		if err := i.configureDirectAPIServerAccess(ctx, cfg, watchLocalNamespace); err != nil {
```

Comment:
This is creating all the secrets before the remote clusters are started which won't guarantee that the correct primary cluster will be associated with each remote cluster. For example, assume we have two primary clusters, p1 and p2, and a remote cluster, r1, which is declared in the topology file to use p1 as its primary. Previously the code used to make sure that p1 controls r1 by:
1. create the r1 secret in p1
2. install r1
3. create the rest of the secrets

Now that you are creating all the secrets at once, both p1 and p2 will have the secret for r1 when it comes up, so they will both compete for control of the cluster, so depending on timing either p1 or p2 might become the primary cluster

Reply:
@frankbu as you have said above, p1 and p2 are primary clusters, secret of r1 will need to be placed on both p1 and p2, regardless when it gets on p1 or p2, timing should not determines if p1 or p2 becomes primary cluster, they are primary clusters when they get setup, I do not think I follow your logic above, you already assumed p1 and p2 are primary clusters, why should r1's secret further determines if p1 or p2 becomes primary cluster?

Reply:
The only way to currently align the bi-directional association (1. remote -> remote-discover-address, 2. primary -> leadership) is by trying to make sure that the declared primary gets the remote secret earlier than other primaries that will compete. It works for the test framework, but I agree it's not reliable since the lock could always get dropped and picked up by another primary. That said, I'm not sure what else we can do for a short term fix.

I am a little confused about why tests don't fail when `remote` has remote discovery address pointing to `primary` but `cross-network-primary` is the leader patching certs in `remote`? I guess it's just because they are compatible certs (same root CA)?

Maybe this is too big a can of worms and we should revert the `feature.ExternalIstiod` to default false and also give up on istiodless defaults for now? Thoughts?

Reply:
This problem is fixed in #39406